includes/functions.php

```diff
@@ -527,3 +543,53 @@ function wbz404_ProcessRedirect($redirect) {
 	}
 }
 
+
+function wbz404_isIPIgnored(){
+  global $wpdb;
+  $ip = esc_sql( wbz404_get_ip_address() );
+  $query = "SELECT IP_address FROM `" . $wpdb->prefix . "wbz404_ignored_IP_addresses` where IP_address = '".$ip."' LIMIT 1";
+  $result = $wpdb->get_results($query);
+  if(count($result) == 1){
+    return true;
+  }
+}
+
+function wbz404_isUserAgentIgnored(){
+  global $wpdb;
+  $user_agent = esc_sql( $_SERVER['HTTP_USER_AGENT'] );
+  $query = "SELECT user_agent FROM `" . $wpdb->prefix . "wbz404_ignored_bots` where user_agent = '".$user_agent."' LIMIT 1";
+  $result = $wpdb->get_results($query);
+  if(count($result) == 1){
+    return true;
+  }
+}
+
+function wbz404_get_ip_address() {
+  # check for shared internet/ISP IP
+  if (!empty($_SERVER['HTTP_CLIENT_IP'])) {
+    return $_SERVER['HTTP_CLIENT_IP'];
+  }
+
+  # check for IPs passing through proxies
+  if (!empty($_SERVER['HTTP_X_FORWARDED_FOR'])) {
+    # check if multiple ips exist in var
+    if (strpos($_SERVER['HTTP_X_FORWARDED_FOR'], ',') !== false) {
+      $iplist = explode(',', $_SERVER['HTTP_X_FORWARDED_FOR']);
+      return $iplist[0];
+    }
+    else{
+      return $_SERVER['HTTP_X_FORWARDED_FOR'];
+    }
+  }
+  if(!empty($_SERVER['HTTP_X_FORWARDED']))
+    return $_SERVER['HTTP_X_FORWARDED'];
+  if(!empty($_SERVER['HTTP_X_CLUSTER_CLIENT_IP']))
+    return $_SERVER['HTTP_X_CLUSTER_CLIENT_IP'];
+  if(!empty($_SERVER['HTTP_FORWARDED_FOR']))
+    return $_SERVER['HTTP_FORWARDED_FOR'];
+  if(!empty($_SERVER['HTTP_FORWARDED']))
+    return $_SERVER['HTTP_FORWARDED'];
+
+  # return REMOTE_ADDR since all else failed
+  return $_SERVER['REMOTE_ADDR'];
+}
```

Comment:
Super nit-picky, but let's match the indentation of the file.  (Turn these spaces into tabs)